src/main/java/org/springframework/data/couchbase/repository/query/CouchbaseQueryMethod.java

```diff
@@ -163,6 +174,24 @@ public String getInlineN1qlQuery() {
 		return StringUtils.hasText(query) ? query : null;
 	}
 
+	/**
+	 * is this a 'delete'?
+	 *
+	 * @return is this a 'delete'?
+	 */
+	public boolean isDeleteQuery() {
```

Comment:
Those methods should go into the actual `CouchbaseQuery` (or `AbstractCouchbaseQuery`). Assuming the method name starts with a particular prefix is pretty risky especially when the query method uses an annotated query.  MongoDB's `@Query` defines attributes such as `delete` or `exists`. `PartTree` (the source for query derivation) exposes similar information.